### PR TITLE
Feat/list items counter

### LINKED ISSUE #40: #17 list-overview-page: list-item counter
## Value Proposition
As a user it would be nice if I'm able to see in the list overview how many list entries are already finished
## Description
user-story-17
## Acceptance Criteria
- [ ] show counter with list-items done / list-items not done
## Tasks
- [x] create feature `feat/list-items-counter`
- [x] add a `counter` state in the list
- [x] update the `counter` state, if any `list-item` gets checked / unchecked / deleted / created

### PR DESCRIPTION
**Pull request by philmetscher**
Heyho. I added the item-counter to each list, to see all completed & uncompleted list-items as a counter

### CODE REVIEW

File: components/category/Category.js [hunk 1]
```diff
@@ -58,10 +58,8 @@ export default function Category({
                 {filteredListItems.map((listItem, index) => (
                   <CategoryListItem
                     key={listItem.id}
-                    id={listItem.id}
+                    listItem={listItem}
                     index={index}
```
Comment (Mark-Krensel) [inline components/category/Category.js:62]: why are you giving the CategoryListItem the index, if you already have an id? I don't know what you intend to with it, but I guess anything you want to do, you may also do with the id. it is also something that doesn't change if you delete items

File: components/lists/Entry.js [hunk 2]
```diff
@@ -6,7 +6,13 @@ import { Draggable } from "@hello-pangea/dnd";
 import { PTag } from "../HtmlComponents";
 import { MdOutlineDragHandle, MdEdit } from "react-icons/md"; //Icons
 
-export default function OverviewListEntry({ children, id, index }) {
+export default function OverviewListEntry({ children, id, index, itemCount }) {
+  const countUncompleted =
+    itemCount[0].toString().length > 1 ? itemCount[0] : `0${itemCount[0]}`;
+  const countCompleted =
+    itemCount[1].toString().length > 1 ? itemCount[1] : `0${itemCount[1]}`;
```
Comment (Mark-Krensel) [inline components/lists/Entry.js]: if you want to always have at least 2 digits you can use ```.padStart()```

File: useStore.js [hunk 3]
```diff
@@ -196,12 +200,21 @@ export const useStore = create(
...
+        const newLists = get().lists.map((list) =>
+          list.id === listItem.listId
+            ? {
+                ...list,
+                itemCount: [
+                  (list.itemCount[0] += listItem.disabled ? -1 : 1),
+                  list.itemCount[1],
+                ],
+              }
+            : list
```
Comment (Mark-Krensel) [inline useStore.js:213]: not sure, but shouldn't L213 be ```: ...list```?

**Review by Mark-Krensel — APPROVED** [review]
Looks good to me, but I never used Zustand, also I don't know how your drag & drop feature works.

**Review by PhilippWeiss85 — APPROVED** [review]

**Pull request merged**